File: software/examples/java/ExampleInput.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletIO16;

public class ExampleInput {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your IO-16 Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletIO16 io = new BrickletIO16(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current value from port A as bitmask
		short valueMaskA = io.getPort('a'); // Can throw com.tinkerforge.TimeoutException
		System.out.println("Value Mask (Port A): " + Integer.toBinaryString(valueMaskA));

		// Get current value from port B as bitmask
		short valueMaskB = io.getPort('b'); // Can throw com.tinkerforge.TimeoutException
		System.out.println("Value Mask (Port B): " + Integer.toBinaryString(valueMaskB));

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/go/example_input.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/io16_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your IO-16 Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	io, _ := io16_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current value from port A as bitmask.
	valueMaskA, _ := io.GetPort('a')
	fmt.Printf("Value Mask (Port A): %b\n", valueMaskA)

	// Get current value from port B as bitmask.
	valueMaskB, _ := io.GetPort('b')
	fmt.Printf("Value Mask (Port B): %b\n", valueMaskB)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
